src/NuGet.Clients/PackageManagement.UI/Models/SearchResultPackageMetadata.cs

```diff
@@ -4,41 +4,161 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Globalization;
 using System.Threading.Tasks;
-using NuGet.Protocol.Core.Types;
 using NuGet.Protocol.VisualStudio;
 using NuGet.Versioning;
 
 namespace NuGet.PackageManagement.UI
 {
+    // This is the model class behind the package items in the infinite scroll list.
+    // Some of its properties, such as Latest Version, Status, are fetched on-demand in the background.
     public class SearchResultPackageMetadata : INotifyPropertyChanged
     {
         private PackageStatus _status;
+
         public event PropertyChangedEventHandler PropertyChanged;
 
+        private static readonly string[] _scalingFactor = new string[] {
+            "",
+            "K", // kilo
+            "M", // mega, million
+            "G", // giga, billion
+            "T"  // tera, trillion
+        };
+
         public string Id { get; set; }
 
+        // Indicates whether the instance is created for solution package manager
+        public bool IsSolution { get; set; }
+
```

Comment:
Property IsSolution is needed because the UI will differ based on whether user is in solution package manager or project package manager.

When in solution package manager, these elements will not be displayed in the infinite list:
- the installed version
- the update button, install button and the uninstall button.